openmdao/core/driver.py

```diff
@@ -655,14 +665,37 @@ def add_constraint(self, name, lower=None, upper=None, equals=None,
         if isinstance(equals, np.ndarray):
             equals = equals.flatten()
 
-        # Scale the lower and upper values
-        if lower is not None:
-            lower = (lower + adder)*scaler
-        if upper is not None:
-            upper = (upper + adder)*scaler
+        # Scale the equality bounds
         if equals is not None:
             equals = (equals + adder)*scaler
 
+        # Scale the upper and lower bounds unless they are infinite.
+        if lower is None:
+            pass
+        elif isinstance(lower, np.ndarray):
+            lower = lower.flatten()
+            # lower is an array, only scale/offset indices > -inf_bound
+            lower = np.where( lower > -inf_bound, (lower + adder)*scaler, -inf_bound )
```

Comment:
Not sure there is really much point in supporting inf as a lower and upper bound in a constraint. If something is less than infinity, then it is better to not add the constraint. Granted, i can see the point on being consistent, but to be totally consistent, then you should also treat 'equals' in the same way, so that if the user does something he never should do, at least it is openmdao inf instead of a larger number.